telcell/celldb/pgdatabase.py

```diff
@@ -0,0 +1,414 @@
+
+
+def _build_cell_identity_query(ci):
+    qwhere = []
+    qargs = []
+    if ci.radio is not None:
+        qwhere.append("radio = %s")
+        qargs.append(ci.radio)
+    if ci.mcc is not None:
+        qwhere.append(f"mcc = {ci.mcc}")
+    if ci.mnc is not None:
+        qwhere.append(f"mnc = {ci.mnc}")
+
+    if isinstance(ci, CellGlobalIdentity):
+        if ci.lac is not None:
+            qwhere.append(f"lac = {ci.lac}")
+        if ci.ci is not None:
+            qwhere.append(f"ci = {ci.ci}")
+
+    elif isinstance(ci, EutranCellGlobalIdentity):
+        if ci.eci is not None:
+            qwhere.append(f"eci = {ci.eci}")
+
+    else:
+        raise ValueError(f"unsupported cell type: {type(ci)}")
+
+    if len(qwhere) == 0:
+        qwhere.append("TRUE")
+
+    return " AND ".join(qwhere), qargs
+
+
+class PgCollection(CellCollection):
+    def __init__(
+        self,
+        con,
+        qwhere=None,
+        qargs=None,
+        qorder=None,
+        count_limit: int = None,
+        on_duplicate: Callable = duplicate_policy.warn,
+    ):
+        """
+        Initializes a `PgDatabase` object.
+
+        The parameter `on_duplicate` specifies the policy for `get()` to apply when the cell database has two or more
+        hits. This may mean that the cell database is inconsistent. The policy is a function that takes a `CellIdentity`
+        and a sequence of cell `Properties` as their two arguments and returns the `Properties` of a single cell or
+        `None`. Currently, the following policies are available:
+        - `celldb.duplicate_policy.drop` (default): returns None
+        - `celldb.duplicate_policy.take_first`: returns the properties of the first cell found
+        - `celldb.duplicate_policy.warn`: same as `take_first`, and emits a warning
+        - `celldb.duplicate_policy.exception`: throws an exception
+
+        @param con: an active postgres connection
+        @param qwhere:
+        @param qargs:
+        @param qorder:
+        @param count_limit:
+        @param on_duplicate: policy when the cell database has two or more hits for the same cell in a call to `get()`.
+        """
+        self._con = con
+        self._qwhere = qwhere or ["TRUE"]
+        self._qargs = qargs or []
+        self._qorder = qorder or ""
+        self._count_limit = count_limit
+        self._cur = None
+        self._on_duplicate = on_duplicate
+
+    def get(self, date: datetime.datetime, ci: CellIdentity) -> Optional[Properties]:
+        """
+        Retrieve a specific antenna from database.
+
+        :param date: Used to select active antennas
+        :param ci: The cell identity
+        :return: The retrieved antenna or None
+        """
+        if isinstance(date, datetime.date):
+            date = datetime.datetime.combine(date, datetime.datetime.min.time())
+
+        qwhere = list(self._qwhere)
+        qargs = list(self._qargs)
+
+        if date is not None:
+            qwhere = qwhere + [
```

Comment:
in de andere repo hebben we hier op een gegeven moment ook nog een filter op azimuth (niet null en <360), celltype (niet indoor) en horizontal_beamwidth (>0 en < 360 of leeg) toegevoegd. Hier ook nog toevoegen?

Reply:
Goed punt; ik wil het nu niet toevoegen omdat het dan te groot wordt, maar heb er wel een issue voor gemaakt